src/server/common/resultsets/game_history.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+   Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#ifdef HAVE_MYSQLPP
+
+#include "server/common/resultsets/game_history.hpp"
+#include "serialization/string_utils.hpp"
+#include "log.hpp"
+
+static lg::log_domain log_sql_handler("sql_executor");
+#define ERR_SQL LOG_STREAM(err, log_sql_handler)
+
+void game_history::read(mariadb::result_set_ref rslt)
+{
+    while(rslt->next())
+    {
+        result r;
+        r.game_name = rslt->get_string("GAME_NAME");
+        r.reload = rslt->get_boolean("RELOAD");
+        r.game_start = rslt->get_date_time("START_TIME").str();
+        r.scenario_name = rslt->get_string("SCENARIO_NAME");
+        r.scenario_id = rslt->get_string("SCENARIO_ID");
+        r.era_name = rslt->get_string("ERA_NAME");
+        r.era_id = rslt->get_string("ERA_ID");
+        for(const auto& player_info : utils::split(rslt->get_string("PLAYERS")))
+        {
+            std::vector<std::string> info = utils::split(player_info, ':');
+            if(info.size() == 2)
+            {
+                r.players.emplace_back(player{ info[0], info[1] });
+            }
+            else
+            {
+                ERR_SQL << "Expected player information to split into two fields, instead found `" << info.size() << "` fields." << std::endl;
```

Comment:
The error should give enough info to find the issue in the DB. Though not sure if we have access to the query data here? Could at least show the offending string then.

Reply:
Only the result set is available here.  I've added the string to the error output.